### PR TITLE
deps: update `tap`

### PR DESCRIPTION
**Pull request by nechaido**
Also remove workaround for previous versions.

### CODE REVIEW

File: test/todo/json5.js [hunk 1]
```diff
@@ -61,14 +60,13 @@ testCases.forEach((testCase) => {
           case '.json5':
             // Currently tap.strictSame() doesn't work well with NaNs.
             // That is why we check if lodash.isequal returns true.
-            test.assert(deepEqual(jstp.parse(file), extendedEval(file)));
             break;
           case '.js': {
             const supportedTests = supportedByUs[testCase.name];
             if (supportedTests && supportedTests.includes(testName)) {
               // Currently tap.strictSame() doesn't work well with NaNs.
               // That is why we check if lodash.isequal returns true.
```
Comment (belochub) [inline test/todo/json5.js]: Shouldn't you remove this comments as well?

Comment (nechaido) [inline test/todo/json5.js]: @belochub sorry, I forgot about it.

File: test/todo/json5.js [hunk 2]
```diff
@@ -59,16 +58,11 @@ testCases.forEach((testCase) => {
             test.strictSame(jstp.parse(file), JSON.parse(file));
             break;
           case '.json5':
-            // Currently tap.strictSame() doesn't work well with NaNs.
-            // That is why we check if lodash.isequal returns true.
-            test.assert(deepEqual(jstp.parse(file), extendedEval(file)));
```
Comment (belochub) [inline test/todo/json5.js]: You forgot to replace it with `strictSame` here.

Comment (nechaido) [inline test/todo/json5.js]: @belochub fixed

**Comment by belochub** [comment]
@nechaido, rebase this on master, please.

**Comment by nechaido** [comment]
Landed in ae2717cb19bdd01cca18e22254d5ccadcb13ce8b.

**Pull request closed without merging**